citro3d/src/util.rs

```diff
@@ -0,0 +1,6 @@
+/// Check if pointer is in linear memory
+pub fn is_linear_ptr<P>(p: *const P) -> bool {
```

Comment:
This does seem useful to have — I've also thought about having some kind of type safety around this, e.g.

```rust
trait Linear {}

impl<T> Linear for Vec<T, LinearAllocator> {}
```

The problem is that other APIs would end up with somewhat inconvenient definitions, e.g.

```rust
    pub fn add<'this, 'a, T, V>(
        &'this mut self,
        vbo_data: &T,
        attrib_info: &attrib::Info,
    ) -> crate::Result<Slice<'a>>
    where
        'this: 'a,
        T: Deref<Target = [V]> + Linear,
    {
```
  
But maybe the type safety would be worth it and then we wouldn't need panics or `Err` types for all these things that require linear allocation.